packages/gnome-shell/src/extensions/prefs.d.ts

```diff
@@ -24,7 +26,7 @@ export class ExtensionPreferences extends ExtensionBase {
      *
      * @param {Adw.PreferencesWindow} window - the preferences window
      */
-    fillPreferencesWindow(window: Adw.PreferencesWindow): void;
+    fillPreferencesWindow(window: Adw.PreferencesWindow): void | Promise<void>;
```

Comment:
Sorry for the back and forth, I didn't notice the type declarations in upstream jsdoc.

But since upstream declares this as Promise<void> I think we should strictly stick to that declaration, and remove the void branch again 😬

Reply:
No problem - Glad we agree on the matter here :). Changed back to `Promise<void>`